datafusion/physical-expr-common/src/sort_expr.rs

```diff
@@ -458,6 +458,21 @@ impl LexOrdering {
         })
     }
 
+    /// Checks if `other` is a prefix of this `LexOrdering`.
+    pub fn is_prefix(&self, other: &LexOrdering) -> bool {
```

Comment:
I meet some similar topic for the sort pushdown, and submitted a [PR](https://github.com/apache/datafusion/pull/19446#issue-3752083820), i prefer to use eq_properties to do similar things, it will handle more cases after my try in my internal project. 


The ordering_satisfy will handle more cases than prefix.

Reply:
I gave it a review - looks nice! I can sequence this behind your PR if needed.